modules/api/test/LorisApiInstrumentsTest.php

```diff
@@ -0,0 +1,933 @@
+                $response->getBody()->getContents()
+            ),
+            true
+        );
+
+        $this->assertSame(gettype($InstrArray), 'array');
+        $this->assertSame(
+            gettype($InstrArray['Meta']),
+            'array'
+        );
+        $this->assertSame(
+            gettype($InstrArray['Meta']['CandID']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($InstrArray['Meta']['Visit']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($InstrArray['Instruments']),
+            'array'
+        );
+        $this->assertSame(
+            gettype($InstrArray['Instruments']['0']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($InstrArray['Instruments']['1']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($InstrArray['Instruments']['2']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($InstrArray['Instruments']['3']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($InstrArray['Instruments']['4']),
+            'string'
+        );
+
+        $this->assertArrayHasKey(
+            'CandID',
+            $InstrArray['Meta']
+        );
+        $this->assertArrayHasKey(
+            'Visit',
+            $InstrArray['Meta']
+        );
+        $this->assertArrayHasKey(
+            '0',
+            $InstrArray['Instruments']
+        );
+        $this->assertArrayHasKey(
+            '1',
+            $InstrArray['Instruments']
+        );
+        $this->assertArrayHasKey(
+            '2',
+            $InstrArray['Instruments']
+        );
+        $this->assertArrayHasKey(
+            '3',
+            $InstrArray['Instruments']
+        );
+        $this->assertArrayHasKey(
+            '4',
+            $InstrArray['Instruments']
+        );
+    }
+
+    /**
+     * Tests the HTTP GET request for the
+     * endpoint /candidates/{candid}/{visit}/instruments{instruments}
+     *
+     * @return void
+     */
+    public function testGetCandidatesCandidVisitInstrumentsInstrument(): void
+    {
+        $response = $this->client->request(
+            'GET',
+            "candidates/$this->candidTest/$this->visitTest/instruments/" .
+            "$this->instrumentTest",
+            [
+                'http_errors' => false,
+                'headers'     => $this->headers
+            ]
+        );
+        if ($response->getStatusCode() === 404) {
+            $this->markTestIncomplete(
+                "Endpoint not found: " .
+                "candidates/$this->candidTest/$this->visitTest/instruments/" .
+                "$this->instrumentTest"
+            );
+        }
+        $this->assertEquals(200, $response->getStatusCode());
+        // Verify the endpoint has a body
+        $body = $response->getBody();
+        $this->assertNotEmpty($body);
+
+        $InstrArray = json_decode(
```

Comment:
first letter of variable name in lowercase

Reply:
right